File: ios/Gekidou/Sources/Gekidou/SessionAttributes/SessionAttributesConstants.swift
//
//  SessionAttributesConstants.swift
//  Gekidou
//
// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
//

import Foundation

public struct SessionAttributesConstants {
    public static let headerName = "X-MM-Session-Attributes"
    public static let storePrefix = "sa_"
    public static let stableValuesKey = "sa_stable_values"
    public static let managedConfigKey = "com.apple.configuration.managed"
    public static let isDeviceManagedKey = "isDeviceManaged"
    public static let isSupervisedKey = "isSupervised"

    public struct AttributeKey {
        public static let vpnActive = "vpn_active"
        public static let clientDeviceId = "client_device_id"
        public static let clientIpAddress = "client_ip_address"
        public static let clientVersion = "client_version"
        public static let jailbreakDetected = "jailbreak_detected"
        public static let mdmEnrolled = "mdm_enrolled"
        public static let networkInterfaceType = "network_interface_type"
        public static let osPlatform = "os_platform"
        public static let osVersion = "os_version"
        public static let serverFqdn = "server_fqdn"
        public static let ssid = "ssid"
    }
}

File: libraries/@mattermost/rnutils/android/src/main/java/com/mattermost/rnutils/session_attributes/SessionAttributesConstants.kt
package com.mattermost.rnutils.session_attributes

object SessionAttributesConstants {
    const val HEADER_NAME = "X-MM-Session-Attributes"
    const val STORE_PREFIX = "sa_"
    const val PREFS_NAME = "session_attributes"
    const val STABLE_VALUES_KEY = "stable_values"
    const val IS_DEVICE_MANAGED_KEY = "isDeviceManaged"
    const val IS_SUPERVISED_KEY = "isSupervised"

    object AttributeKey {
        const val VPN_ACTIVE = "vpn_active"
        const val CLIENT_DEVICE_ID = "client_device_id"
        const val CLIENT_IP_ADDRESS = "client_ip_address"
        const val CLIENT_VERSION = "client_version"
        const val JAILBREAK_DETECTED = "jailbreak_detected"
        const val MDM_ENROLLED = "mdm_enrolled"
        const val NETWORK_INTERFACE_TYPE = "network_interface_type"
        const val OS_PLATFORM = "os_platform"
        const val OS_VERSION = "os_version"
        const val SERVER_FQDN = "server_fqdn"
        const val SSID = "ssid"
    }
}
